lib/core/src/Cardano/Wallet/Primitive/Types/TokenBundle.hs

```diff
@@ -330,6 +334,14 @@ getQuantity = TokenMap.getQuantity . tokens
 hasQuantity :: TokenBundle -> AssetId -> Bool
 hasQuantity = TokenMap.hasQuantity . tokens
 
+-- | Sets the quantity associated with a given asset.
+--
+-- If the given quantity is zero, the resultant bundle will not have an entry
+-- for the given asset.
+--
+setQuantity :: TokenBundle -> AssetId -> TokenQuantity -> TokenBundle
```

Comment:
```suggestion
setQuantity :: AssetId -> TokenQuantity -> TokenBundle -> TokenBundle
```

Reply:
Suggestion: for easier currying and to make it's nature as a transformation on `TokenBundle` more obvious.

Reply:
i.e. `setQuantity` is also a "transformation" like `mapAssetIds`.

Reply:
I can definitely sympathise with the currying argument.

Though tor `TokenBundle.setQuantity`, I've chosen to mirror the signature of `TokenMap.setQuantity`, which uses the same argument order.

We could change them both so that they follow the order in your suggestion, but we'd have to change quite a lot of client code.